### ISSUE TITLE
Is these patchs work with Xorg ?

### ISSUE THREAD

**Issue by makotoworkshop** [D0023R/linux_kernel_15khz#2]
Hello !

Thank you for continue working on the 15kHz patchs ^^

I've successfully compile somes kernels, and it's work when it is enable in the selection of the desired video mode during the boot process (boot loader grub), but in the unique resolution of "640x480", in console command line mode (320x240 and 800x600 not working).

- _Unfortunately when Xorg is started, the display not working in 15 kHz_.
(Same problem when it is not enable in boot loader)
- Is these patchs work with Xorg ?
- If yes, what kind of mistake i haved doing ?

…

**Here is my compile routine :**
```
################################
# compilation de kernel 15 KHz #
################################


1 — installer debian 10 Buster :
    ——————————————————————————

	apt install build-essential bc kmod cpio flex libncurses5-dev dpkg-dev debconf-utils debhelper fakeroot zlib1g-dev rsync


2 — Préparatifs :
    ———————————

    Le récupérer les sources et les extraire dans un dossier de travail : /home/user/kernel5.0
	mkdir /home/user/kernel5.0
	cd /home/user/kernel5.0
	wget https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.0.1.tar.gz
	tar xvf linux-5.0.1.tar.gz

    Se rendre dans le dossier crée :
	cd linux-5.0.1

    Configurer le kernel :

	make olddefconfig

    Cette commande va chercher la configuration du kernel actuel (booté) et met toutes les nouvelles options en « par défaut »

    Editer le fichier .config, afin de vérifier que cette ligne ne contient rien entre les guillemets (sinon la compilation plantera dés les premières minutes) :
    
       CONFIG_SYSTEM_TRUSTED_KEYS = ""

    Exécuter ce script permet de ne pas compiler un noyau de debug, et donc de gagner du temps :

         ./scripts/config -d CONFIG_DEBUG_INFO


4 — Patcher :
    ———————

    Récupérer les patchs 15kHz :
	cd /home/user/
	wget https://github.com/D0023R/linux_kernel_15khz/archive/master.zip

    Les extraires :
	unzip master.zip

    Tester les patchs :
	cd /home/user/kernel5.0/linux-5.0.1/
	patch -p1 --dry-run < ../../linux_kernel_15khz-master/linux-5.0/01_ati_9200_pllfix.diff
	patch -p1 --dry-run < ../../linux_kernel_15khz-master/linux-5.0/02_arcadevga_3000.diff
	patch -p1 --dry-run < ../../linux_kernel_15khz-master/linux-5.0/03_linux_15khz.diff
	patch -p1 --dry-run < ../../linux_kernel_15khz-master/linux-5.0/04_linux_15khz_scanoutpos.diff
	patch -p1 --dry-run < ../../linux_kernel_15khz-master/linux-5.0/05_linux_15khz_interlaced_mode_fix.diff

    Patcher les sources :
	patch -p1 < ../../linux_kernel_15khz-master/linux-5.0/01_ati_9200_pllfix.diff
	patch -p1 < ../../linux_kernel_15khz-master/linux-5.0/02_arcadevga_3000.diff
	patch -p1 < ../../linux_kernel_15khz-master/linux-5.0/03_linux_15khz.diff
	patch -p1 < ../../linux_kernel_15khz-master/linux-5.0/04_linux_15khz_scanoutpos.diff
	patch -p1 < ../../linux_kernel_15khz-master/linux-5.0/05_linux_15khz_interlaced_mode_fix.diff


5 - Compilation du kernel :
    —————————————————————
	make -j$(nproc) bindeb-pkg LOCALVERSION=-"$(dpkg --print-architecture)"-patched15khz

    Après plusieurs minutes/heures, on obtiens alors ces packets :
	linux-5.0.1-amd64-patched15khz_5.0.1-amd64-patched15khz-1_amd64.buildinfo
	linux-5.0.1-amd64-patched15khz_5.0.1-amd64-patched15khz-1_amd64.changes
	linux-headers-5.0.1-amd64-patched15khz_5.0.1-amd64-patched15khz-1_amd64.deb
	linux-image-5.0.1-amd64-patched15khz_5.0.1-amd64-patched15khz-1_amd64.deb
	linux-image-5.0.1-amd64-patched15khz-dbg_5.0.1-amd64-patched15khz-1_amd64.deb
	linux-libc-dev_5.0.1-amd64-patched15khz-1_amd64.deb
    (Sauf le packet « -dbg_ » si la compilation du noyau débug a été désactivée)


6 - Installation du kernel :
    ——————————————————————
	dpkg -i	linux-image-5.0.1-amd64-patched15khz_5.0.1-amd64-patched15khz-1_amd64.deb
```

**Here is my _Xorg.conf_ working great with my old Debian Jessie Kernel 3.2.55 Patched 15kHz :**

```
Section "Device"
    Identifier  "ATI"
    Driver      "radeon"
EndSection

Section "Monitor"
        Identifier   "TV"
        HorizSync    15.0 - 20.0
        VertRefresh  50.0 - 60.0
## modelines TV @15kHz
    Modeline "320x240" 6.452 320 338 368 412 240 242 245 264  -HSync -VSync
EndSection

Section "Screen"
    Identifier "Default Screen"
    Device     "ATI"
    Monitor    "TV"
    DefaultDepth     24
    SubSection "Display"
    Depth     24
    Modes    "320x240"
    EndSubSection
EndSection
```

**Comment by substring** [comment]
Salut @makotoworkshop ;)

we hope to release a new version of the 15k patch with kernel 5.5 that would require **no X configuration** (you can go xorg.conf-less). The problem with the current set of patches is that DRM rejects the resolution (too long to explain) and doesn't make it available to X. This has been fixed on my side. Need some more testing. If you feel like giving a go to the new patches, just tell me.

**Comment by makotoworkshop** [comment]
Hi substing !
Thank you for theses explanations.
Yes i would happy to tests theses new patches ^^

**Comment by substring** [comment]
Here is the current state of the patch. It's a replacement for patch 03. I must admit I simply patched my kernel with this patch and didn't apply all the others. I guess it should work fine though.

Kernel command line changed a bit. Forget the 15/25 switches, just set a resolution like before, add ``i`` for interlace, ``S`` is mandatory as its the switch that triggers the patch code. As before, the code will look through an array of fixed resolutions, the interlace switch does matter.

Here are some examples:
```
video=320x240S
video=640x480iS
```

Needless to say the video default kernel options still work. This patch was tested on Linux 5.4, I hope it can be backported, no idea.

With this patch you shouldn't need to configure Xorg, please confirm :smiley: 
[drm_switchres.zip](https://github.com/D0023R/linux_kernel_15khz/files/4050457/drm_switchres.zip)

**Comment by substring** [comment]
By the way this is still beta until Doozer adds it to his repo.

**Comment by makotoworkshop** [comment]
Hello Subs !

I've successfully patch the kernel found here : https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.4.tar.gz with this patch list :
> 01
> 02
> _drm_switchres.patch_
> 04
> 05

- But the compilation failed !
- I've tried to patch with only _drm_switchres.patch_, but same problem.

make[2]: *** [debian/rules:6: build]  Error2
dpkg-buildpackage: erreur: debian/rules build subprocess returned exit status2
make[1]: *** [scripts/Makefile.package:83: bindeb-pkg] Error 2
make: *** [Makefile:1430: bindeb-pkg] Error 2

**Comment by makotoworkshop** [comment]
Hello Subs !

Sorry for the delay, but i experience some issues with compiling a kernel simple (without any patch)
I've successfully compiled kernel 5.0, 5.1 and 5.2, but since 5.3, the compilation fail  :
(with my routine described above, and others methods i've found)
```
make[2]: *** [debian/rules:6: build]  Error2
dpkg-buildpackage: erreur: debian/rules build subprocess returned exit status2
make[1]: *** [scripts/Makefile.package:83: bindeb-pkg] Error 2
make: *** [Makefile:1430: bindeb-pkg] Error 2
```
So sorry, i can't yet patch a kernel 5.4 with drm_switchres.patch.
I'll happy to take from you a cheat or method for solve that problem, in the meantime, i continue to seach and understood wath happening with those kernels.

**Comment by substring** [comment]
Hey !

I'm really not familiar with Debian packaging system (tried it a while ago, despicable), and if Debian has a linux mainline package or not. You may eventually try on a 5.2 kernel, it could work.

**Comment by makotoworkshop** [comment]
Ok, i'll tried that on next monday.
Thanks ^^

**Comment by makotoworkshop** [comment]
Hi Subs !

The reason that make failed, is that i've forget to remove some text between quotes on the .config file… My bad, i've forget to write this on my remind notes, and unfortunately i forget this quickly. So i've edit my first post to append that note about the line (CONFIG_SYSTEM_TRUSTED_KEYS = "")

For information the _drm_switchres.patch_ not working without errors messages on kernel 5.2.

I've successfully patch the kernel found here : https://cdn.kernel.org/pub/linux/kernel/v5.x/linux-5.4.tar.gz with this patch list :
-     01_(5.4 version)
-     02_(5.4 version)
-     drm_switchres.patch
-     04_(5.4 version)
-     05_(5.4 version)

I'll test it on my Arcade system as soon as possible, with something like that to the boot loader :
`append root=/dev/sda1 rw vga=785 <...other parameters...> video=VGA-1:320x240Sec`

**Comment by substring** [comment]
@makotoworkshop Warning: commandline parameters have changed ! You should rather use `video=320x240eS` (prefer `video=640x480iS` for interlaced 640x480), the 15/25kHz switches have been unified in a single S(witchres) switch.

**Comment by substring** [comment]
I meant `video=640x480eiS`. `c` and `y` switches have been removed

**Comment by makotoworkshop** [comment]
I'm back with some tests results.

In the file _/etc/default/grub_, i replace the working line that for the old kernel 5.2 patched  :
`GRUB_CMDLINE_LINUX="video=DVI-I-1:640x480ec"`
( _note for some unknow reason, xrandr say that the display out is DVI-0, but that not working... I've found that i've to write DVI-I-1 instead !_ )

by this line for the kernel 5.4 patched with drm_switchres béta :
`GRUB_CMDLINE_LINUX="video=DVI-I-1:640x480eiS"`

then
`update-grub`

then 
`reboot`

- The command line interface appear on my Arcade Monitor 15kHz :)
- If i run STARTX with a 640x480i modeline, X appear on the Monitor :)

Of course 640x480eS mode, is not working.

But, the mode who is the best is the 320x240p...
So **i've tried 320x240eS, but it's Nothing** on my Arcade Monitor 15kHz :(


For be sure, i've re-tested the kernel 5.2 patched 15kHz, and yes, 320x240ec not working, and, as i've talk in last october, 640x480ec work in command line interface,
 but i've say that X does not work, and it's a mistake, i think in october i've run a progressive modeline with, cause when i run STARTX now with a 640x480 interlaced modeline, it's work !


So, in conclusion at the moment, the behaviour of my system with any version of patchs cause the same issue :

- 320x240 NOT working in command line interface or X interface (320x240p modeline)
- 640x480i working in command line interface or X interface (640x480i modeline)

Hope this will help you for your future release !
If you need more tests i can do it.

Thanks

**Comment by makotoworkshop** [comment]
I will tried to upgrade the OS of my Arcade system and redo theses tests.
I remind this OS is might be old and that can have some impact on the results.

**Comment by substring** [comment]
Whenever you are testing, please report the contents of `/sys/class/drm/card0-DVI-I-1/{status, enabled,modes}` (note `modes`, we'll need it later). For the X problem, `modes` should be filled with the resolution you required and use it natively (DRM will handle the full modeline to X). If X refused it but the framebuffer is fine, check `modes` and make sure the resolution is available, also check the X log. I'll test 320x240 on my side too.

What GFX card are you using ? which driver are you using (open source/proprietary ? radeon/amdgpu ?) I hope it's not the X GPU driver rejecting the resolution.

640x480eS can't work for 2 reasons:
- it's a 31kHz modeline, so probably a bad idea on a TV or arcade monitor
- resolutions are hard coded for now, and progressive 640x480 is not part of it (see patch lines 105 to 146).

**Comment by makotoworkshop** [comment]
Hello Subs,

Thanks for that explanations !
My  first tests where's silly guided by the old way with manual modeline.

### My hardware and driver config is : 
```
lspci
01:00.1 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] RV515 GL [FireGL V3350] (Secondary)
```

```
lsmod | grep radeon
radeon               1638400  2
ttm                   122880  1 radeon
drm_kms_helper        212992  1 radeon
drm                   548864  5 drm_kms_helper,radeon,ttm
i2c_algo_bit           16384  1 radeon
```
A card that still work very well with my old patched 3.22 Kernel and manual 320x240 modeline.

So here the tests i've made today, on a fresh installed  up to date OS : 
### 1. _With **Kernel 5.2** patched 15kHz_ : 

- Grub config : "video=DVI-I-1:**640x480ec**"
> Display of command line interface Ok on the CRT screen !

> But, **_startx_** nothing on the CRT screen.
```
cat /sys/class/drm/card0-DVI-I-1/{status,enabled,modes}
connected
enabled
```

.

- Grub config : "video=DVI-I-1:**320x240ec**"
> Get Nothing on the CRT screen

.
.
.

### 2. _With **Kernel 5.4** patched 15kHz with **drm_switchres béta** :_ 

- Grub config : "video=DVI-I-1:**640x480eiS**"
Display of command line interface Ok on the CRT screen !
> **_startx_**  Display X on the CRT screen ! Your _drm_switchres_ work :+1: 
```
cat /sys/class/drm/card0-DVI-I-1/{status,enabled,modes}
connected
enabled
640x480i
```
```
xrandr
Screen 0: minimum 320 x 200, current 640 x 480, maximum 8192 x 8192
DVI-0 connected primary 640x480+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   640x480i      60.00* 
DVI-1 connected 640x480+0+0 (normal left inverted right x axis y axis) 359mm x 287mm
   1280x1024     60.02 +
   1280x960      60.00  
   1024x768      75.03    70.07    60.00  
   800x600       72.19    75.00    60.32  
   640x480       75.00    72.81    59.94* 
   720x400       70.08  
```

.

- Grub config : "video=DVI-I-1:**320x240eS**"
> Get **Nothing** on the CRT screen.

> And, **_startx_** nothing on the CRT screen too.
```
cat /sys/class/drm/card0-DVI-I-1/{status,enabled,modes}
connected
enabled
320x240
```
```
xrandr
Screen 0: minimum 320 x 200, current 1280 x 1024, maximum 8192 x 8192
DVI-0 connected primary 320x240+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   320x240       60.00* 
DVI-1 connected 1280x1024+0+0 (normal left inverted right x axis y axis) 359mm x 287mm
   1280x1024     60.02*+
   1280x960      60.00  
   1024x768      75.03    70.07    60.00  
   800x600       72.19    75.00    60.32  
   640x480       75.00    72.81    59.94  
   720x400       70.08  
```

### Same conclusions as yesterday.

**Comment by substring** [comment]
Tried on my side :

- dmesg reports `[drm] Found a switchres mode for 320x240 (interlace: 0)` which is fine
- DRM is perfect too:
```
[arcade@GroovyArcade ~]$ cat /sys/class/drm/card0-VGA-1/{status,enabled,modes}
connected
enabled
320x240
```
- Xorg:
```
[    23.896] (II) RADEON(0): Printing probed modes for output VGA-0
[    23.896] (II) RADEON(0): Modeline "320x240"x60.0    6.64  320 336 368 424  240 242 245 261 -hsync -vsync (15.7 kHz e)
[    23.896] (II) RADEON(0): Output HDMI-0 disconnected
[    23.896] (II) RADEON(0): Output DVI-0 disconnected
[    23.896] (II) RADEON(0): Output VGA-0 connected
[    23.896] (II) RADEON(0): Using exact sizes for initial modes
[    23.896] (II) RADEON(0): Output VGA-0 using initial mode 320x240 +0+0
```
- xrandr:
```
[arcade@GroovyArcade ~]$ DISPLAY=:0.0 xrandr
Screen 0: minimum 320 x 200, current 320 x 240, maximum 8192 x 8192
HDMI-0 disconnected primary (normal left inverted right x axis y axis)
DVI-0 disconnected (normal left inverted right x axis y axis)
VGA-0 connected 320x240+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   320x240       60.00*
```
So from my point of view, it's running perfect. Can you link you Xorg log ? I think it will give us some details about possible problems.

But it's wierd, you said 320x240 console worked, but in your last comment you said it didn't, whereas DRM reported fine 320x240. I'm lost :laughing:

**Comment by makotoworkshop** [comment]
I've never said @320x240 console worked, or if you have read this, it might be my poor english level in cause, sorry.

Effectively, it's weird,  _/sys/class/drm/card0-DVI-I-1/{status,enabled,modes}_  and _xrandr_ tells that's ok, but really, nothing displayed on the CRT. (except the highly blurred image, typical of an invalid too high frequency)

Here the complete [Xorg.log](https://github.com/D0023R/linux_kernel_15khz/files/4096415/Xorg.log)

**Comment by substring** [comment]
Bon, arrangeons-nous des problèmes de langue (j'ai déjà lu ton blog :smile: )

Tu aurais 2 écrans connectés en DVI (l'arcade et un autre de debug j'imagine). X a bien trouvé le 320x240:
```
[    20.318] (II) RADEON(0): Modeline "320x240"x60.0    6.64  320 336 368 424  240 242 245 261 -hsync -vsync (15.7 kHz e)
```
Autre preuve:
```
DVI-0 connected primary 320x240+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   320x240       60.00* 
```
Mais le mode 320x240 ne marche pas du tout que ce soit en console ou sur X, que ce soit avec l'ancien patch ou le nouveau.

Je te demanderais bien 2 tests distincts (qui impliquent de désactiver `video=...`):
- utiliser cette modeline pour X
```
	ModeLine          "320x240x60.00" 6.639840 320 336 368 424 240 242 245 261 -HSync -VSync
```
- forcer un EDID pour DVI-I-1 avec le fichier attaché à déposer dans /us/lib/firmware/edid. Ajoute comme paramètre au noyau `drm.edid_firmware=DVI-I-1:edid/generic_15.bin`

Si la modeline ne passe pas, ca sera vraiment bizarre. Eventuellement tester sans brancher le 2e moniteur (dès le boot), qui sait.

**Comment by makotoworkshop** [comment]
Ha bah oui, ce sera plus simple :+1: 
J'espère que je ne raconte pas trop de bêtises sur mon blog ^^;

J'ai donc effectivement 2 écrans connectés en DVI analogique (donc au final c'est du VGA), le premier pour le CRT et l'autre en LCD informatique.

Donc oui, le mode 320x240 ne marche pas du tout que ce soit en console ou sur X, quelque soit la version du patch, sur l'écran de ma borne et idem sur une TV 36 cm, que j'ai branché hier sans circuit de protection électronique (https://burogu.makotoworkshop.org/index.php?post/2011/10/02/borne-arcade-7), ce qui a permis de constater l'image brouillée, indiquant un modeline non supporté.

* Je vais donc tester le modeline que tu indiques sans paramètrage vidéo dans grub , et j'en profitterais pour retester aussi celui que j'utilisais jusqu'à présent et qui est un peu différent.

* Pour le deuxième test en forçant EDID dans grub, je ne vois pas de fichier attaché à ton post, ou alors je ne sais pas où regarder. 

* Pour le troisième test, sans brancher le second moniteur dés le boot, c'est une idée à essayer car ce serait bien le genre de truc qui se produit sans qu'on comprenne bien pourquoi.

À priori je ferais ça Jeudi soir.

**Comment by substring** [comment]
[generic_15.zip](https://github.com/D0023R/linux_kernel_15khz/files/4098987/generic_15.zip)

La modeline est pour X ;)

Oui j'ai raté l'EDID, je le posterai ce soir une fois rentré.

Pour les 2 moniteurs, vu qu'ils partagent le même framebuffer, ca pourrait être une explication. Je ne vois pas pourquoi 320x240 marcherait chez moi et pas chez toi, sachant qu'avec le nouveau patch, tout est bien déclaré comme il faut, la résolution est bien listée par l'OS.

Edit: EDID inclus, à décompresser.

**Comment by makotoworkshop** [comment]
J'ai juste le temps ce soir de tester la modeline et le second écran débranché.
çà n'a pas fonctionné pour ces deux tests.
Ci joint le [xorg.conf.txt](https://github.com/D0023R/linux_kernel_15khz/files/4099024/xorg.conf.txt) et les [Xrandr320.txt](https://github.com/D0023R/linux_kernel_15khz/files/4099035/Xrandr320.txt) effectués chaques fois (et chaque fois écran noir sur le CRT)
Je vais voir si j'ai le temps de passer le fichier edid

**Comment by makotoworkshop** [comment]
Voilà, j'ai passé vite fait l'edid, mais sans voir plus avant si le fichier généric était bien pris en compte par le systéme, il me semble qu'on l'apperçoit dans [Xorg.0.log](https://github.com/D0023R/linux_kernel_15khz/files/4099096/Xorg.0.log)
* çà n'a rien donné à l'écran :,(

Si on ne peut pas se baser sur le log pour valider que cet edid est pris en compe, je regarderais demain pour en avoir une preuve, et sinon pour faire en sorte qu'il le soit, en espérant que çà fonctionne ^^;
Je pourrais aussi t'envoyer mon kernel si jamais çà peut etre utile.
Merci, bonne soirée !

**Comment by substring** [comment]
Il est pris en compte:
```
[    13.823] (II) RADEON(0): Monitor name: generic_15
```

**Comment by substring** [comment]
Faut pas que le kernel, faut aussi la clique des modules, et pour tout ce qui est arcade je suis sur Arch Linux. Je peux éventuellement te faire une version de GroovyArcade avec ce noyau en attendant.

**Comment by makotoworkshop** [comment]
Ok si  le fichier edid est bien pris en compte… alors je suppose qu'on arrive au bout des tests envisageables ?
Je n'ai jamais utilisé Arch, c'est peut-être l'occasion de voir ça. Il va me falloir quelques jours je pense.

* Si j'installe une Arch, il faudra que je lui compile un noyau patché et si tout fonctionne avec mon matériel, on pourra accuser Debian…
* Où sinon accuser le hardware; tout au moins son non-support par le patch, pour une raison à déterminer.
* Je pourrais tester aussi si je trouve un autre modèle de  carte vidéo, dés fois que…

Merci de ta proposition, j'y aurais recours si je n'arrive à rien avec Arch, mais j'imagine que je devrais m'en sortir  (enfin l'espère ^^;)

**Comment by substring** [comment]
Question bête : ça marchait avant avec cette carte en 320x240 ? Idem pour ton moniteur ?

Sinon oui on est bien au bout des tests qui me viennent en tête. J'ai déjà le noyau patché pour Arch, je peux te l'envoyer et te donner les consignes pour l'installer/le prendre en compte. Si je t'ai parlé de GroovyArcade, c'est parce que c'est la distribution historique de GroovyMAME que j'essaie un peu de ressusciter et remettre à jour. Je peux te faire un iso avec le kernel patché déjà embarqué. On évitera de trop digresser sur cette issue par contre, ca déborde du sujet original :wink:

Par contre, je vais réintégrer les e et y en paramètres de video= pour la rétro compatibilité, ton cas a été utile !

Dernière question quand même : je ne sais pas quel émulateur (mame ? groovymame ? autre ?) tu utilises, mais si tu bootes en 640x480i et que tu lances un jeu en 240p avec groovymame, le moniteur switche bien de résolution ? Peut-être tu n'es pas allé jusque là ?

**Comment by makotoworkshop** [comment]
* Oui ça fonctionn·ait·e très bien en 320x240, avec exactement le même matériel.
Dés que je rebranche mon ancien disque, sur-lequel est installé le noyau 3.2 que j'avais patché à l'époque (que vous avez linké dans le Readme ici même), j'ai l'image nikel, avec chacun des modelines 320x240 de mon xorg.conf.

En fait je suis revenu à la recherche de patchs récents car je souhaite upgrader ma borne.
En effet, le mame 0.208 que j'avais compilé ne supporte plus SDL1.2, et donc le passage en SDL2 m'oblige avec _Debian_ à mettre à jour l'OS, mais cette fois c'est le noyaux qui se révèle trop vieux…

Comme j'avais déjà mis en place mon système avant(?) la découverte(/création) de _GroovyMAME_, et n'étant pas familier de _ArchLinux_, j'avais décidé de simplement utiliser le combo « _Debian_ + kernel15khz + Xorg bloqué sur 320x240p + Mame officiel ».
Donc non, je n'utilise pas _GroovyMAME_. Si tu souhaites des tests, indique moi la ressource pour le télécharger ou le compiler, et je fais ça ^^

Autrement ok, je veux bien ton noyau et les consignes pour le mettre sur Arch (au moins j'apprendrais des choses), ça me fera gagner pas mal de temps, car j'aimerais bien tirer ça au clair, et savoir si le soucis du 240p est dû à une subtilité incompatible avec _Debian_ ou autre chose qui cloche.

Content de savoir ces échanges auront été utiles, bravo pour l'initiative de « réssucitage » ! Pour avoir moi aussi entrepris de générer mes propres iso live, je sais que ya du taf !

**Comment by D0023R** [comment]
Hi,
Switching back to English. I just released the new 15kHz patch from Substring into the master branch. 
@makotoworkshop have you managed to figure out what's going wrong on your side?

**Comment by makotoworkshop** [comment]
Hello _D0023R_  !

To sum up, with the great help of _Substring_, we have beta-testing the new patch, and after a lot of efforts, can conclude that the GPU i've used (FireGL3350) have a low-pixelclock intolerance. 
The Groovymame dotclock_min option was not very helpful, unfortunately.

I've decided to change my GPU for a Radeon 7540, but keep in mind that this hardware is ok to work with my old configuration with the kernel 3.22 patched 15kHz (Calamity patch from 2013)

I want again to Thank your team (and special for _Substring_ ^^) for continue working on theses patchs and others retro-gaming hardware compatibility projects.

greetings all

**Comment by D0023R** [comment]
Thank you makotoworkshop for the testing and nice words. I am not sure that saying that I own a team is correct in the present context. I am pleased to get support from the community and I really appreciate the help of Substring. Congrats to both of you for having investigated the issue.
